README.md

```diff
@@ -1,25 +1,104 @@
+├── requirements.txt            # For dependencies installation
+├── student_list.csv            # A mapping from ccids to github usernames
+├── token_file.txt              # Your Github access token(You need to create this file)
+├── update_repos.sh             # Update all repos
+│
+├── assignment1                 # A subdirectory named with an assignment or lab name 						
+│   │                           # It's automatically created after running collect_repos.py
+│   ├── repo1 
+│   │   ├── feedback.txt        # Feedback you made
+│   │   └── ...
+│   ├── repo2                   # Student's repository	
+│   └── ...                
+└── ...                         # Other subdirectories (e.g., labs, assignment2, etc.)
+```
+
+
+
+## Instructions
+
+To use these scripts, please follow the following steps:
+
+1. **Generate a personal access token**	
+
+   - Go to https://github.com/settings/tokens, click `Generate new token`, select all scopes and click `Generate token`.
+
+   - Remember to copy the token and paste it to a file created by yourself named `token_file.txt`.
+
+2. **Configure**
+
+   - Open `config`, make your own configuration. Typically, you need to change `username`, `prefix` and `deadline`.
+   - `prefix` is the name of lab or assignment you are grading (e.g., assignment1).
+   - If you want to collect all repos, provide no value to key `list`.
+   - If you want to collect repos with the latest commit, provide no value to key `deadline`. Otherwise, the script finds the closest commit that happens before a given deadline (inclusive) and clones/checks out the repository at that commit.
+
+3. **Add ccids**
+
+   - Open `ccid_list.txt`, paste ccids you need to grade to this file.
+   - All ccids are emails (e.g., zicun@ualberta.ca), one ccid per line.
 
-The scripts have been tested with Python 3.9.1(latest).
+4. **Collect repositories**
 
-# Script descriptions:
+   ```shell
+   # Run this command in the root directory.
+   
+   Python3 collect_repos.py
+   
+   # If you are grading assignment1, 
+   # a directory called 'assignment1' will be created, 
+   # where all the repos for assignment1 will be cloned to.
+   ```
 
-Create a directory called `assignmentName` where all the repos for this assignment will be cloned to. You can run any of the following scripts from within that directory.
+5. **Grade**
 
-- `collect_assignment_at_deadline.py`: For a given assignment, it finds the closest commit that happens before a given deadline (inclusive) and clones/checks out the repository at that commit. Run `python3 collect_assignments_at_deadline.py --help` to see required arguments.
+6. **Push feedback to student's repository**
 
-- `collect_assignments_now.py`: same as above but gets the current state of the repo (can be useful if you will pull all repos at the deadline yourself)
+   ```shell
+   # If your feedback is in 'feedback.txt', run this command in the root directory.
+   
+   ./push_feedback.sh feedback.txt
```

Comment:
Actually one prob I just thought of. For labs, this will be Lab1/feedback.txt. I assume it would still work right?

Reply:
Great catch! I just submitted some changes to make it work.